src/Stratis.Bitcoin/Broadcasting/BroadcasterManagerBase.cs

```diff
@@ -58,5 +60,25 @@ public void AddOrUpdate(Transaction transaction, State state)
         }
 
         public abstract Task<Success> TryBroadcastAsync(Transaction transaction);
+
+        protected void PropagateTransactionToPeers(Transaction transaction, bool skipHalfOfThePeers = false)
+        {
+            this.AddOrUpdate(transaction, State.ToBroadcast);
+
+            var invPayload = new InvPayload(transaction);
+
+            var propagateTo = skipHalfOfThePeers
```

Comment:
Is it not easier to just look at the count of connected nodes and do a `for` loop that goes to i < nodes.length/2?

Reply:
Yeah. It's unnecessary over-complicated. Fixing.